semantic_router/index/pinecone.py

```diff
@@ -313,8 +323,15 @@ def add(
             self.index = self._init_index(force_create=True)
 
         vectors_to_upsert = [
-            PineconeRecord(values=vector, route=route, utterance=utterance).to_dict()
-            for vector, route, utterance in zip(embeddings, routes, utterances)
+            PineconeRecord(
+                values=vector,
+                route=route,
+                utterance=utterance,
+                function_schema=str(function_schema),
```

Comment:
in what scenario would `function_schema` not be a string already? If `function_schema` is a `dict` type, should we be using `json.dumps(function_schema)` here?